torch/_export/__init__.py

```diff
@@ -575,6 +577,7 @@ def _export(
     constraints: Optional[List[Constraint]] = None,
     *,
     preserve_module_call_signature: Tuple[str, ...] = (),
+    disable_constraint_solver: bool = False,
```

Comment:
@avikchaudhuri @gmagogsfm @aakhundov did we have discussion around this? I'm not recently working on dynamic shape and I don't remember we've talked about letting people disable the solver.

Reply:
So the issue we're hitting with an internal model export (as part of the internal AOT Inductor lowering) is that the errors like this are generated for constraints like in the unit test added in this PR: `s1 == s0 + 1` (you can see that if you set `disable_constraint_solver=False` in the unit test added in this PR):

```
  - b = L['getattr_10'].size()[0] must be specialized to 32 because the guards generated for it are too complex.
```

Other examples of constraints that cause an error:

    - `Eq(512*(((2*s1)//s0)), 1024)`
    - `512*(((2*s1)//s0)) >= 1024`.

As this is relatively urgent, I was thinking that adding a flag to circumvent the constraint solver to the non-public API (`torch._export`, `torch._dynamo`) would be acceptable as a short-term solution. Which we could then address more fundamentally (like extending the constraint solver to deal with the constraints that lead to errors currently). If you think we should dig into the constraint solver and can find and fix the root cause in a short amount of time, I'm happy to go down that route.